### ISSUE TITLE
False positive on Performance/RegexpMatch

### ISSUE THREAD

**Issue by ckruse** [bbatsov/rubocop#5038]
Performance/RegexpMatch false detects code like this to be a valid candidate for using `.match()` instead of `=~`:

~~~ruby
return Regexp.last_match(1).to_i if uri =~ %r{/m?(\d+)$}
~~~

## Expected behavior

Rubocop should correctly identify the use of `Regexp.last_match`

## Actual behavior

Rubocop suggests to replace `=~` by `.match()`

## Steps to reproduce the problem

Run Rubocop against the code above, with Ruby version 2.4.2:

~~~
➜ ckruse@spinnweb ~/sites/cforum (master) ✗ % cat .ruby-version
2.4.2
➜ ckruse@spinnweb ~/sites/cforum (master) ✗ % cat test.rb
# frozen_string_literal: true

return Regexp.last_match(1).to_i if uri =~ %r{/m?(\d+)$}
➜ ckruse@spinnweb ~/sites/cforum (master) ✗ % rubocop test.rb
Inspecting 1 file
C

Offenses:

test.rb:3:37: C: Use match? instead of =~ when MatchData is not used.
return Regexp.last_match(1).to_i if uri =~ %r{/m?(\d+)$}
                                    ^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
➜ ckruse@spinnweb ~/sites/cforum (master) ✗ %
~~~

## RuboCop version

```
➜ ckruse@spinnweb ~/sites/cforum (master) ✗ % rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-darwin17)
➜ ckruse@spinnweb ~/sites/cforum (master) ✗ %
```

**Comment by pocke** [comment]
https://github.com/bbatsov/rubocop/blob/bf3edd8f66ad71bfcefc0f87413eb5087b702b16/lib/rubocop/cop/performance/regexp_match.rb#L156

This cop should not use `loc.expression.begin_pos` at here. But I don't know what we should use instead.

**Comment by Mifrill** [comment]
```
C: Performance/RegexpMatch: Use match? instead of =~ when MatchData is not used.
      next if category =~ skipped_categories
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
